src/NHibernate.Validator/Constraints/DecimalMaxAttribute.cs

```diff
@@ -98,6 +99,9 @@ public bool IsValid(object value, IConstraintValidatorContext validatorContext)
 
 		public void Apply(Property property)
 		{
+			if (AttributeUtils.AttributeUsedMultipleTimesOnProperty(property, GetType()))
+				return;

```

Comment:
This seems to be done for all `IPropertyConstraint` with `AllowMultiple`. Then better move that check into `ClassValidator`, for not calling this method at all when constraints should not be generated (when the attribute allows multiple and is defined multiple times). By the way, this should allow to fix the component case, since `ClassValidator` has access to more data about the property.